pynws/forecast.py

```diff
@@ -71,15 +103,13 @@ def last_update(self) -> datetime:
     @staticmethod
     def _find_detail_for_time(
         when, time_values: tuple[datetime, datetime, Any], units: str | None
-    ) -> tuple[Any, str | None]:
+    ) -> Value | None:
         for start_time, end_time, value in time_values:
             if start_time <= when < end_time:
-                return value, units
-        return None, None
+                return Value(value, units)
```

Comment:
The value/unit tuple has been replaced with an object that can do unit conversions.